### ISSUE TITLE
Content of response with 205 HTTP status code should be empty

### ISSUE THREAD

**Issue by sunghyunzz** [akka/akka-http#2685]
IMHO, response with 205 HTTP status code should have empty content. I checked following two documents:
1. http://en.wikipedia.org/wiki/List_of_HTTP_status_codes
1. https://tools.ietf.org/html/rfc7231#section-6.3.6

However current version of Akka HTTP sends contents while its HTTP status code is 205. 
```
HTTP/1.1 205 Reset Content
Content-Length: 80
Content-Type: text/plain; charset=UTF-8
Date: Fri, 06 Sep 2019 06:07:31 GMT
Server: akka-http/10.1.7

The server successfully processed the request, but is not returning any content.
```
It could make trouble when other HTTP client libraries do not expect contents to be sent. (for example, iOS app uses [Alamofire](https://github.com/Alamofire/Alamofire/blob/538edbfb891ab154f638a598e657b75c73d7e9ec/Source/ResponseSerialization.swift#L104) would be in a trouble)

**Comment by jrudolph** [comment]
I agree, thanks for the PR.

**Comment by jrudolph** [comment]
Fixed in #2686.
